hack/generator/pkg/functions/pivot_conversion_function.go

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package functions
+
+import (
+	"fmt"
+
+	"github.com/dave/dst"
+
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astbuilder"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/conversions"
+)
+
+// PivotConversionFunction implements a pivot that's used when a conversion reaches a hub type but hasn't yet found the
+// innermost required conversion needed.
+//
+// We use this for Spec types, implementing genruntime.ConvertibleSpec, and for Status types, implementing
+// genruntime.ConvertibleStatus.
+//
+// If the two instances involved in conversion are not in the same "spoke" leading to this "hub" version, we'll pivot to
```

Comment:
I got confused and asked you about this and you explained it very nicely with a diagram, then you said you should add some ascii art to this.

Reply:
Added comment and diagram.

Reply:
A link to the design doc on this section with a non-ascii diagram would also work

Reply:
Although I see you've got the ascii diagram already so maybe nevermind =)